src/core/ddsi/src/ddsi_ipaddr.c

```diff
@@ -101,10 +102,9 @@ enum ddsi_locator_from_string_result ddsi_ipaddr_from_string (const struct ddsi_
   // hostnames seem to be limited to 255 characters, add a port of max 5
   // digits and a colon, and so 262 should be enough.  (Numerical addresses
   // add a few other characters, but even so this ought to be plenty.)
-  const size_t len = strlen(str);
-  if (len == 0 || len >= sizeof(copy))
+  size_t cnt = ddsrt_strlcpy(copy, str, sizeof(copy));
```

Comment:
It seems MSVC likes your code as much as you like MSVC :wink:

Reply:
Haha! Then MSVC really has a very strong dislike of me 🙂

It is not even love lost, the only C compilers I have used voluntarily that I can remember are Turbo C (and its successors), Watcom, gcc (+ egcs, long live egcs!), the SUN, SGI and DEC compilers, clang of course, also Analog Devices' (I found I could *easily* beat its optimizer but their assembler unfortunately was more than happy to accept certain invalid inputs and assemble them into "something"). Not to forget xlC and Intel's. And Acorn's (see http://chrisacorns.computinghistory.org.uk/Computers/ACW.html) and https://bellard.org/tcc/, of course! I dare say I am pretty easy, but the abomination that MSVC is I have only ever used because it was either that or going hungry. 

Strangely, despite my complete utter and total dislike of MS's compiler, I do have respect for their compiler division; and despite that dislike, I suppose I am one of those people who found Windows 3.0 quite useful because of its use of virtual 8086 mode, reducing the time I had to spend rebooting my PC :)

Once OS/2 3.0, Solaris x86, Linux 2.0 showed up, I had no use for Windows anymore, except for 1 year before switching to MacOS. And every time I have had to deal with it since then, it took only like 15 minutes before I was ready to end my career there and then and start growing papayas in Iceland or something impossible like that 😁